external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -115,4 +119,25 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationSuite with V2JDBCTest {
     val expectedSchema = new StructType().add("ID", IntegerType, true, defaultMetadata)
     assert(t.schema === expectedSchema)
   }
+
+  override def testIndex(tbl: String): Unit = {
+    val loaded = Catalogs.load("mysql", conf)
+    val jdbcTable = loaded.asInstanceOf[TableCatalog]
+      .loadTable(Identifier.of(Array.empty[String], "new_table"))
+    assert(jdbcTable.asInstanceOf[SupportsIndex].indexExists("i1") == false)
```

Comment:
`jdbcTable.asInstanceOf[SupportsIndex]` occurs many times. Maybe create a variable for it and reuse below.

Reply:
Actually all occurrence of `jdbcTable` below are used as `SupportsIndex`. So you can do
```scala
val jdbcTable = loaded.asInstanceOf[TableCatalog]
  .loadTable(Identifier.of(Array.empty[String], "new_table"))
  .asInstanceOf[SupportsIndex]
```